.github/workflows/ci.yml

```diff
@@ -11,44 +11,73 @@ jobs:
     name: build
 
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        img: ["ubuntu:20.04", "ubuntu:22.04", "centos:8" ]
+      fail-fast: true
+    container:
+      image: ${{ matrix.img }}
 
     steps:
     - uses: actions/checkout@v3
-    - name: apt update
-      run: sudo apt-get update
-    - uses: awalsh128/cache-apt-pkgs-action@latest
-      with:
-        packages: xfsprogs gnuplot net-tools numactl libnuma-dev xfslibs-dev bc time strace
-        version: 1.0
+
+    - name: Ubuntu Prerequisites
+      if: startsWith(matrix.img, 'ubuntu:')
+      run: contrib/CI/ubuntu.sh
+
+    - name: CentOS 8 Prerequisites
+      if: matrix.img == 'centos:8'
+      run: contrib/CI/centos8.sh
+      
     - name: configure
       run: cmake -B ${{github.workspace}}/build
     - name: make
       run: make -C ${{github.workspace}}/build
     - name: make install
-      run: sudo make -C ${{github.workspace}}/build install
+      run: make -C ${{github.workspace}}/build install
     - name: run tests
-      run: ctest --output-on-failure
-      working-directory: ${{github.workspace}}/build
+      run:  | 
+            cd ${{github.workspace}}/build 
+            ctest --output-on-failure
```

Comment:
Why change this?

Reply:
I ran into errors using the working-directory syntax in the workflow. I am not too sure why it was causing issues, but swapping to this fixed it. 
![image](https://github.com/MigeljanImeri/xdd/assets/78048439/5f87aecb-34f0-4e4a-b0df-ba5f8e4c1f36)

Reply:
Ok. Just double checking that that was the issue. The github actions variables are weird.